core/controller/src/main/scala/whisk/core/controller/Meta.scala

```diff
@@ -44,27 +156,35 @@ trait WhiskMetaApi extends Directives with PostActionActivation {
     /** Store for identities. */
     protected val authStore: AuthStore
 
-    /** The route prefix e.g., /experimental/package-name. */
-    protected val routePrefix = pathPrefix("experimental")
+    /** The route prefix e.g., /experimental. */
+    protected val routePath = "experimental"
+
+    /** The prefix for anonymous invokes e.g., /experimental/web. */
+    protected val anonymousInvokePath = "web"
 
     /** The name and apikey of the system namespace. */
     protected val systemId = "whisk.system"
     protected lazy val systemNamespace = EntityPath(systemId)
     protected lazy val systemKey = WhiskAuth.get(authStore, Subject(systemId), false)(TransactionId.controller)
 
-    /** Reserved parameters that requests may no defined. */
-    protected lazy val reservedProperties = Set("__ow_meta_verb", "__ow_meta_path", "__ow_meta_namespace")
+    private val routePrefix = pathPrefix(routePath)
+    private val anonymousInvokePrefix = pathPrefix(anonymousInvokePath)
 
     /** Allowed verbs. */
     private lazy val allowedOperations = get | delete | post
 
-    /** Extracts the HTTP method, query params and unmatched (remaining) path. */
+    private lazy val validNameSegment = pathPrefix(EntityName.REGEX.r)
+    private lazy val packagePrefix = pathPrefix("""\Adefault\z""".r | EntityName.REGEX.r)
```

Comment:
I believe `"default".r` is sufficient and equivalent to `"""\Adefault\z""".r` as regexes are anchored in Scala by default.